mergebot_poller.py

```diff
@@ -1,47 +1,126 @@
+  CMDS = { 'merge': CMD_MERGE }
+  CMD_MERGE = { 'github': merge_git }
+
+  def __init__(self, config):
+    self.config = config
 
+  def poll(self):
+    """
+    Kicks off polling of the SCM. Reads SCM type from config and starts the
+    appropriate poller.
 
-  def poll(self, config):
-    print('Starting merge poller for {}.'.format(config['name']))
-    self.scm_type = config['scm_type']
-    self.repository = config['repository']
-    self.branch = config['default_branch']
-    scm_pollers[scm_type](config)
+    :return: returns nothing
+    """
+    print('Starting merge poller for {}.'.format(self.config['name']))
+    SCM_POLLERS[self.config['scm_type']]()
 
+  def poll_github(self):
+    """
+    Polls Github for PRs, verifies they haven't already been worked on, and
+    then searches them for merge-bot commands.
 
-  def poll_github(self, config):
-    # Loop: Forever, once per minute.
+    :return: returns nothing
+    """
+    # Instantiate the two variables used for tracking work.
+    # Thread-safe because of the GIL.
+    self.work_queue = Queue.Queue()
+    self.known_work = set()
+    # Set up a github helper for handling network requests, etc.
+    self.github_helper = GithubHelper(ORG, self.config['repository'])
+    # Kick off an SCM merger
+    merger = Thread(target = SCM_MERGERS['github'],
+                    args = (self.work_queue, self.known_work, self.github_helper, ))
+    merger.start()
+    # Loop: Forever, every fifteen seconds.
     while True:
-      print('Loading pull requests from Github at {}.'.format(PULLS_URL))
       # Load list of pull requests from Github
-      r = requests.get(PULLS_URL, auth=(BOT_NAME, bot_key))
-      if r.status_code != 200:
-        print('Oops, that didn\'t work. Error below, waiting then trying again.')
-        print(r.text)
-        return
-
-      print('Loaded.')
-      pr_json = r.json()
-      # Loop: Each pull request
-      for pr in pr_json:
-        search_pr(pr)
-
+      prs = self.github_helper.fetch_prs()
+      if prs is not None:
+        # Loop: Each pull request
+        for pr in prs:
+          if pr['number'] not in self.known_work:
+            search_github_pr(pr)
+      else:
+        print('Error fetching pull requests.')
       time.sleep(15)
 
+  def search_github_pr(self, pr):
+    """
+    Searches a provided github PR for mergebot commands, validates, and runs
+    said commands.
+
+    :param pr: The pull request to validate
+    :return: returns nothing
+    """
+    pr_num = pr['number']
+    print('Looking at PR #{}.'.format(pr_num))
+    # Load comments for each pull request
+    cmts = self.github_helper.fetch_comments(pr_num)
+    if len(cmts) < 1:
+      print('No comments on PR #{}. Moving on.'.format(pr_num))
+      return
+    # FUTURE: Loop over comments to make sure PR has been approved by a
+    # committer.
+    cmt = cmts[-1]
+    cmt_body = cmt['body'].encode('ascii', 'ignore')
+    # Check auth.
+    user = cmt['user']['login']
```

Comment:
This seems to check the auth of every user that posted a comment. Perhaps we should first skip comments that don't look like a command, so that we don't complain about every user (and stop searching) just because non-committers made comments?